pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -188,15 +197,24 @@ func (r *SubnetPortReconciler) deleteSubnetPortByName(ctx context.Context, ns st
 		return err
 	}
 
+	var hasSubnetPort bool
+	var externalIpAddress *string
 	for _, nsxSubnetPort := range nsxSubnetPorts {
 		if crSubnetPortIDsSet.Has(*nsxSubnetPort.Id) {
 			log.Info("skipping deletion, SubnetPort CR still exists in K8s", "ID", *nsxSubnetPort.Id)
+			hasSubnetPort = true
 			continue
 		}
+		if nsxSubnetPort.ExternalAddressBinding != nil && nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress != nil && *nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress != "" {
+			externalIpAddress = nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress
```

Comment:
in which case *nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress = “”？

Reply:
It's just a validation, we cannot guarantee that NSX response is always valid.